shell/platform/linux/fl_renderer_wayland.cc

```diff
@@ -0,0 +1,322 @@
+// Copyright 2020 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#define WL_EGL_PLATFORM
+
+#include "fl_renderer_wayland.h"
+#include "flutter/shell/platform/linux/egl_utils.h"
+
+#include <gdk/gdkwayland.h>
+#include <wayland-egl-core.h>
+
+static void fl_renderer_wayland_on_window_map(GtkWidget* widget,
+                                              FlRendererWayland* self);
+static void fl_renderer_wayland_on_window_unmap(GtkWidget* widget,
+                                                FlRendererWayland* self);
+
+struct _FlRendererWayland {
+  FlRenderer parent_instance;
+
+  struct {
+    wl_subsurface* subsurface;
+    wl_surface* surface;
+    wl_egl_window* egl_window;
+    GdkRectangle geometry;
+    gint scale;
+  } subsurface;
+
+  // The resource surface will not be mapped, but needs to be a wl_surface
+  // because ONLY window EGL surfaces are supported on Wayland.
+  struct {
+    wl_surface* surface;
+    wl_egl_window* egl_window;
+  } resource;
+};
+
+static wl_registry* registry_global = nullptr;
```

Comment:
Why are these global? We're only likely to have one FlRenderer so we don't need to optimise this.

Reply:
As long as there generally wont be multiple renders than the renderer can own them. Something will certainly need need to be changed for multi-window support (right now a renderer can only have one window).